UIKit library fails to build on Linux

```
In file included from /home/kamit/Downloads/plexydesk-master/base/uikit/core/desktopactivity.h:7:0,
                 from /home/kamit/Downloads/plexydesk-master/base/uikit/core/view_controller.h:9,
                 from /home/kamit/Downloads/plexydesk-master/base/uikit/core/extensionmanager.h:28,
                 from /home/kamit/Downloads/plexydesk-master/base/uikit/themepackloader.cpp:29:
/home/kamit/Downloads/plexydesk-master/base/uikit/core/widget.h:84:10: error: ‘std::function’ has not been declared
     std::function<void (InputEvent a_type,
          ^
/home/kamit/Downloads/plexydesk-master/base/uikit/core/widget.h:84:18: error: expected ‘,’ or ‘...’ before ‘<’ token
     std::function<void (InputEvent a_type,
                  ^
base/uikit/CMakeFiles/PlexyDeskUICore.dir/build.make:77: recipe for target 'base/uikit/CMakeFiles/PlexyDeskUICore.dir/themepackloader.cpp.o' failed
make[2]: *** [base/uikit/CMakeFiles/PlexyDeskUICore.dir/themepackloader.cpp.o] Error 1
CMakeFiles/Makefile2:140: recipe for target 'base/uikit/CMakeFiles/PlexyDeskUICore.dir/all' failed
make[1]: *** [base/uikit/CMakeFiles/PlexyDeskUICore.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/base/uikit/core/widget.h
+++ b/base/uikit/core/widget.h
@@ -23,6 +23,8 @@
 #include <style.h>
 #include <plexydesk_ui_exports.h>
 
+#include <functional>
+
 #include <QGraphicsLayoutItem>
 #include <QGraphicsObject>
 

--- a/base/uikit/imagecache.h
+++ b/base/uikit/imagecache.h
@@ -21,11 +21,13 @@
 #ifndef IMAGE_CACHE_H
 #define IMAGE_CACHE_H
 
-#include <QtCore/QObject>
-#include <QtGui/QPainter>
-
 #include <plexy.h>
 #include <plexyconfig.h>
+
+#include <functional>
+
+#include <QObject>
+#include <QPainter>
 
 #include "plexydeskuicore_global.h"
 #include <plexydesk_ui_exports.h>

